metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -116,6 +122,9 @@ public void execute(Tuple tuple) {
       boolean ackTuple = !writer.handleAck();
       int numWritten = 0;
       if(sensorParserConfig != null) {
+        if (configUpdatedFlag.getAndSet(false)) {
+          parser.configurationUpdated(getSensorParserConfig().getParserConfig());
```

Comment:
Ah.  All that's needed is to fetch sensorParserConfig AFTER doing configUpdatedFlag.getAndSet(false).  So right before line 118 do

  boolean mustUpdateConfig = configUpdatedFlag.getAndSet(false); //must check before fetching sensorParserConfig

then in the current line 125 use mustUpdateConfig instead of configUpdatedFlag.getAndSet(false).